extensions/mssql/src/tableDesigner/tableDesigner.ts

```diff
@@ -6,21 +6,23 @@
 import { AppContext } from '../appContext';
 import * as azdata from 'azdata';
 import * as vscode from 'vscode';
-import { sqlProviderName, TableType } from '../constants';
+import { sqlProviderName } from '../constants';
 import { generateUuid } from 'vscode-languageclient/lib/utils/uuid';
 import { ITelemetryEventProperties, Telemetry } from '../telemetry';
 
 export function registerTableDesignerCommands(appContext: AppContext) {
 	appContext.extensionContext.subscriptions.push(vscode.commands.registerCommand('mssql.newTable', async (context: azdata.ObjectExplorerContext) => {
 		const connectionString = await azdata.connection.getConnectionString(context.connectionProfile.id, true);
-		const telemetryInfo = await getTelemetryInfo(context, TableType.Basic);
+		const tableSubType = context.nodeInfo.nodeSubType as azdata.designers.TableSubType ?? azdata.designers.TableSubType.Temporal;
```

Comment:
I wouldn't call it tableSubType, it is nodeSubType. I'd call it tableType. this also applies to all other places.

Reply:
actually, let's call it iconName, graph/temporal/memory optimized are not table types, just features of the table, some tables might 2 of these features at the same time.

Reply:
Hm, although I think we want to have this be usable for stuff other than just the icon. You're going to want to display different options for the different type of table right? Unless you have access to these properties somewhere else (in which case why not just use that?)

Reply:
Thinking about it a bit more having a parameter for the icon makes sense to me since we want any other provider to be able to use this - so it shouldn't be scoped to just SQL table types.

But that means we should be having the extension calling this pass in the icon to use, not having core do that logic. (although I suppose we could add support for generic codicons as well)

Reply:
the purpose of this new property is solely to help the core to determine the icon, I don't want to call it table type because it is not an accurate definition.

Reply:
yes, ideally we should allow extension authors to provide icon. but in ADS, the available icons for editor is defined in a centralized location: https://github.com/microsoft/azuredatastudio/blob/main/extensions/theme-seti/icons/vs-seti-icon-theme.json#L1821

Reply:
Yeah, so that's what I mean by the SqlThemeIcon. Any "centralized" icons should ideally be exposed through that instead of having hard-coded logic like this to map them. So for anything that wants to use those icons for this (or anything else) they can just use that, or they can pass in their own icon.